src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -3343,12 +3370,37 @@ def list_artifacts(
             query = select(ArtifactSchema)
             if artifact_uri is not None:
                 query = query.where(ArtifactSchema.uri == artifact_uri)
+            if artifact_store_id is not None:
+                query = query.where(
+                    ArtifactSchema.artifact_store_id == artifact_store_id
+                )
             artifacts = session.exec(query).all()
             return [
                 self._artifact_schema_to_model(artifact)
                 for artifact in artifacts
             ]
 
+    def delete_artifact(self, artifact_id: UUID) -> None:
+        """Deletes an artifact.
+
+        Args:
+            artifact_id: The ID of the artifact to delete.
+
+        Raises:
+            KeyError: if the artifact doesn't exist.
+        """
+        with Session(self.engine) as session:
```

Comment:
Apparently, this is already not consistent. Currently, only stack, stack component, and flavor have their deletion implemented like this and all other entities use `first()` and raise a `KeyError` manually, similar as we do here. Personally, I don't see any benefit in doing this one way rather than another, so I'd just stick with the current implementation.